website/src/utils/search.ts

```diff
@@ -175,81 +148,82 @@ export const consolidateGroupedFields = (filters: MetadataFilter[]): Consolidate
             fieldList.push(filter);
         }
     }
-
     return fieldList;
 };
 
-export const getFieldValuesFromQuery = (
-    state: Record<string, string>,
-    hiddenFieldValues: FieldValues,
-    schema: Schema,
-): FieldValues => {
-    const values: FieldValues = { ...hiddenFieldValues };
-    const expandedSchema = getMetadataSchemaWithExpandedRanges(schema.metadata);
-    for (const field of expandedSchema) {
-        if (field.name in state) {
-            values[field.name] = state[field.name];
-        }
-    }
-    if ('accession' in state) {
-        values.accession = state.accession;
-    }
-    if ('mutation' in state) {
-        values.mutation = state.mutation;
-    }
-    return values;
-};
-
-const textAccessionsToList = (text: string): string[] => {
-    const accessions = text
-        .split(/[\t,;\n ]/)
-        .map((s) => s.trim())
-        .filter((s) => s !== '')
-        .map((s) => {
-            if (s.includes('.')) {
-                return s.split('.')[0];
-            }
-            return s;
-        });
+/**
+ * Derives from the Metadata schema. For some metadata fields, they are expanded into multiple
+ * (grouped) filters.
+ */
+export class FilterSchema {
```

Comment:
I'm wondering if this should have `Metadata` in the name too?

Reply:
sure!

https://github.com/loculus-project/loculus/pull/3883/commits/b8348b1fabac1f66280c8f4ef6ccae2cfb002902